Updated GSL orographic drag suite to enable use of custom orographic …

mdtoy opened the pull request:
This is an update of the GSL orographic drag suite to use separate "small-scale" and "large-scale" orographic drag parameters (varss, olss, etc.), both of which can be read in by custom created static files.  These are modified regression testing files that I used to test the code.

Note that this pull request (along with two forthcoming PR's in the fv3atm and ccpp-physics submodules) will cause the loss of functionality with the EMC ugwp drag suite.   Some additional coding will be needed to be able to switch between the two drag suites.  There are two new compile-time variables, NEW_ORO_STAT" and "NEW_ORO_STAT_INFO", which change some of the compiled code.  I hope you don't mind my leaving some of the final coding decisions to you.  You folks know the FV3 code much better than I do. But I'd be happy to try if you'd rather I come up with something. Please let me know.

mdtoyNOAA committed:
Updated GSL orographic drag suite to enable use of custom orographic statistics static files

mdtoy commented:
Sorry, I forgot to mention something.  Currently, the RT for the updated drag suite links static files located in my "mtoy" directory on Hera.  Feel free to copy these oro_data files to a more permanent location.  Thanks.

climbfuji commented on tests/fv3_conf/ccpp_gsd_run_drag_suite.IN:
```diff
@@ -0,0 +1,100 @@
+mkdir INPUT RESTART
+if [ $WARM_START = .F. ]; then
+  cp -r @[RTPWD]/FV3_input_data_gsd/FV3_input_data_C96_with_aerosols/* INPUT/
+  ln -s /scratch2/BMC/wrfruc/mtoy/FV3GFS/GWD_static_files/C96/tile1/oro.tile1.nc ./INPUT/oro_data_ls.tile1.nc
```
I will need to copy these files into the regression test baseline and ask you to change the paths here. Also, we could add a check in `ccpp_gsd_run.IN` if the GSD drag suite is used (similar to how the switches are made for the microphysics/lsm) instead of duplicating the entire file?

DomHeinzeller replied:
Thanks for adding this logic to `tests/fv3_conf/ccpp_gsd_run.IN`. I will remove tests/fv3_conf/ccpp_gsd_run_drag_suite.IN when I pull your code into the latest version of gsd/develop (happening now).

climbfuji commented on tests/tests/fv3_ccpp_gsd_drag_suite:
```diff
@@ -84,7 +84,7 @@ export DO_SAT_ADJ=.F.
 export LTAEROSOL=.T.
 
-export FV3_RUN=ccpp_gsd_run.IN
+export FV3_RUN=ccpp_gsd_run_drag_suite.IN
```
No need to change this if we can add a switch in the existing `ccpp_gsd_run.IN`

DomHeinzeller replied:
Thanks for making the change to `ccpp_gsd_run.IN`. I will revert this change when I pull your code into the latest version of gsd/develop (happening now).

mdtoyNOAA committed:
Merged with latest NOAA-GSD/ufs-weather-model gsd/develop branch.

mdtoyNOAA committed:
Merging latest changes to GSL drag suite with current NOAA-GSD/ufs-weather-model gsd/develop branch.

Merge branch 'gsd/develop' of https://github.com/NOAA-GSD/ufs-weather-model into updated_GSL_drag_suite_apr2020
